### ISSUE TITLE
Reconsider typescript export to cover both esModuleInterop modes and esm by default

### ISSUE THREAD

**Issue by rosskevin** [i18next/i18next#1351]
We originally had `export default i18next` (which was still a namespace from previous @types) before https://github.com/i18next/i18next/pull/1291, then we switched to `export = i18next` with a namespace.  @PierreAndreis and I both agreed this to be the right solution.  In order to mitigate any fallout, we added tests for both `esModuleInterop` states, of which the default is `false`.

After #1291 change, we have had a larger number of typescript import issues (#1342, #1316, #1350, not to mention others in the i18next-react project), so I took a look at what it meant to get this running, thanks in part to @sraisty1c putting together a repo in https://github.com/i18next/react-i18next-gitbook/issues/63.

What I found and documented were the gory details of what is necessary to get a webpack+jest project both type checking and running.  It took extensive knowledge of the loading semantics and was quite painful.  In large part this is because nodejs uses our `cjs` (commonjs) package, and webpack uses our `esm` package.  Node/jest only loads cjs, whereas webpack loads `esm` then falls back to `cjs` in that order by default.

---

I spent four hours on this and documented my findings yesterday in https://github.com/rosskevin/test-i18next/tree/non-esmoduleinterop-ts-loader

---

More than not, even with the default `esModuleInterop: false`, it seems like the current scheme is too painful.  It is strange that typescript has not changed the default.   As stated in the [typescript 2.7 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-from-commonjs-modules-with---esmoduleinterop), they recommend running __**with**__  `--esModuleInterop` even though it is still not the default:

>Note: The new behavior is added under a flag to avoid unwarranted breaks to existing code bases. We highly recommend applying it both to new and existing projects. For existing projects, namespace imports (import * as express from "express"; express();) will need to be converted to default imports (import express from "express"; express();).

## Conclusion

I have always wanted to get rid of the namespace, but I recall @PierreAndreis and I having problems when attempting to do that https://github.com/i18next/i18next/pull/1286#pullrequestreview-263857803.  It seems that we opted there for a `cjs` first approach, and this has caused more pain then it relieved.

Quite simply, the index is an `export default`.  Yes we have both cjs and esm modules, but it seems that given most of our user base is using `esm` and it is the way forward, let's examine flipping it back to `export default` without the namespace and if there is mitigation to be required, do so for older `cjs` situations, not now and future `esm` situations.  

## TL;DR

This seems more difficult than it has to be.  `export default` is a modern `esm` semantic, let's support it first out of the box for everyone and require interop for cjs if we must (as recommended by [typescript 2.7 release notes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-from-commonjs-modules-with---esmoduleinterop))

I'm going to try removing the namespace again given this knowledge, and see if I can move it forward and understand the implications to both cjs and esm users.

**Comment by Skillz4Killz** [comment]
@rosskevin It seems this has somehow broken my code when updating to the latest 19.0. Not sure if im doing something wrong or if something perhaps isnt supported in the 19.0 version.

```ts
import i18next from 'i18next'
console.log('testing', i18next)
```
![image](https://user-images.githubusercontent.com/23035000/68531673-e40ad100-02e2-11ea-95b6-bf8fbeea835c.png)

Did i miss some step or something silly of me?

Details: I do not enable esModuleInterop because by default its disabled. I do not use ts-node nor jest which were the two breaking issues in the PR mentioned.

**Comment by rosskevin** [comment]
That is no reason to leave it disabled! `esModuleInterop` is disabled by default for non-breking legacy purposes! Typescript recommends it! Please read the link I posted https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-7.html#support-for-import-d-from-cjs-from-commonjs-modules-with---esmoduleinterop

Now, since it doesn't work for you, that means you are using typescript source code BUT loading the legacy `cjs` or commonjs code package. Either change your tooling to load the `esm` package OR enable `esModuleInterop` AS IS RECOMMENDED by the typescript team.

Sorry to be so direct, but your question asks me to restate the facts from both this issue and the PR, so I need to be more forceful it seems.
